app/battles.py

```diff
@@ -0,0 +1,18 @@
+from app.knight import Knight
+
+
+def one_pair_battle(knight1: Knight, knight2: Knight) -> dict:
+    knight1.hp -= knight2.power - knight1.protection
+    knight2.hp -= knight1.power - knight2.protection
+    return {knight1.name: max(knight1.hp, 0),
+            knight2.name: max(knight2.hp, 0)}
+
+
+def final_battle(pairs: list) -> dict:
+    total_scores = {}
+    for pair in pairs:
+        name1, name2 = pair
+        knight1 = Knight.knights_instances[name1]

```

Comment:
Ensure that `Knight.knights_instances` is a correctly initialized and populated dictionary in the `Knight` class. This is crucial for accessing knight instances by their names.